centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -483,6 +483,28 @@ object Operations extends StrictLogging {
     } yield selected
   }
 
+  def validateSubworkflowsMetadataBeforeAndAfterArchival(submittedWorkflow: SubmittedWorkflow,
+                                                         workflow: Workflow): Test[Unit] = new Test[Unit] {
+
+    def validateSubworkflowsMetadata(expected: WorkflowFlatMetadata, actual: WorkflowFlatMetadata): IO[Unit] = {
+      val errors = expected diff actual
+      if (errors.isEmpty) {
+        IO.pure(())
+      } else {
+        val message = s"Subworkflows metadata validation failed: ${errors.mkString("\n")}"
+        IO.raiseError(CentaurTestException(message, workflow, submittedWorkflow))
+      }
+    }
+
+    override def run: IO[Unit] = {
+      for {
+        notArchivedMetadata <- CentaurCromwellClient.metadata(submittedWorkflow, expandSubworkflows = true, archived = Option(false))
+        archivedMetadata <- CentaurCromwellClient.metadata(submittedWorkflow, expandSubworkflows = true, archived = Option(true))
+        _ <- validateSubworkflowsMetadata(notArchivedMetadata.asFlat.onlySubWorkflows, archivedMetadata.asFlat.onlySubWorkflows)
```

Comment:
I suggest we use the `diffson` tool to generate real diffs of the metadata json (rather than indirectly via the `asFlat` method). You can hopefully extract and reuse the `JsObject` comparison method from the `def validateJobManagerStyle` method in this same file.